packages/cli/src/commands/package-dynamic-plugins/command.ts

```diff
@@ -0,0 +1,229 @@
+    ) {
+      Task.log(
+        `Running export-dynamic-plugin on plugin package ${packageFilePath}`,
+      );
+      try {
+        await Task.forCommand(`yarn export-dynamic`, { cwd: packageDirectory });
+      } catch (err) {
+        Task.log(
+          `Encountered an error running 'yarn export-dynamic on plugin package ${packageFilePath}, this plugin will not be packaged.  The error was ${err}`,
+        );
+      }
+    } else {
+      Task.log(
+        `Using existing 'dist-dynamic' directory at ${packageDirectory}`,
+      );
+    }
+  }
+  // create temporary directory to hold staged plugins for packaging
+  const tmpDir = fs.mkdtempSync(
+    path.join(os.tmpdir(), 'package-dynamic-plugins'),
+  );
+  const pluginRegistryMetadata = [];
+  try {
+    // copy the dist-dynamic output folder for each plugin to some temp directory and generate the metadata entry for each plugin
+    for (const pluginPkg of packages) {
+      const {
+        packageDirectory,
+        packageFilePath,
+        packageJson,
+        packageRoleInfo,
+      } = pluginPkg;
+      const packageName =
+        packageJson.name.replace(/^@/, '').replace(/\//, '-') +
+        ((packageRoleInfo && packageRoleInfo.platform) === 'node'
+          ? '-dynamic'
+          : '');
+
+      const distDirectory = path.join(packageDirectory, 'dist-dynamic');
+      const targetDirectory = path.join(tmpDir, packageName);
+      Task.log(`Copying '${distDirectory}' to '${targetDirectory}`);
+      try {
+        fs.cpSync(distDirectory, targetDirectory, { recursive: true });
+        const {
+          name,
+          version,
+          description,
+          backstage,
+          homepage,
+          repository,
+          license,
+          maintainers,
+          author,
+          bugs,
+          keywords,
+        } = packageJson;
+        pluginRegistryMetadata.push({
+          [packageName]: {
+            name,
+            version,
+            description,
+            backstage,
+            homepage,
+            repository,
+            license,
+            maintainers,
+            author,
+            bugs,
+            keywords,
+          },
+        });
+      } catch (err) {
+        Task.log(
+          `Encountered an error copying static assets for plugin ${packageFilePath}, the plugin will not be packaged.  The error was ${err}`,
+        );
+      }
+    }
+    // Write the plugin registry metadata
+    const metadataFile = path.join(tmpDir, 'index.json');
+    Task.log(`Writing plugin registry metadata to '${metadataFile}'`);
+    await fs.writeFile(
+      metadataFile,
+      JSON.stringify(pluginRegistryMetadata, undefined, 2),
+    );
+    // run the command to generate the image
+    Task.log(`Creating image using ${containerTool}`);
+    await Task.forCommand(
+      `echo "from scratch
+COPY . .
+" | ${containerTool} build --annotation com.redhat.rhdh.plugins='${JSON.stringify(pluginRegistryMetadata)}' -t '${tag}' -f - .
```

Comment:
Is there a way to check if the containerTool is available before building plugins? A fail fast approach: "The tool containerTool is not installed, please make sure it is installed before running this command".